tests/cupy_tests/core_tests/test_ndarray_copy_and_view.py

```diff
@@ -335,6 +335,37 @@ def test_copy_host_to_device_view(self):
             dev[:] = host
 
 
+class TestViewDtype:
+
+    @testing.numpy_cupy_array_equal()
+    def test_smaller_dtype_multiple(self, xp):
+        # x is non-contiguous
+        x = xp.arange(10, dtype=xp.int32)[::2]
+        with pytest.raises(ValueError):
+            x.view(xp.int16)
+        return x[:, xp.newaxis].view(xp.int16)
```

Comment:
The `shape[-1] == 1 and stride[-1] == 0 != itemsize` case is covered by this testcase.

[Optional] Add another testcase to cover other `shape[-1] == 1 and stride[-1] != itemsize case`, for example `xp.ones((3, 4), xp.int32)[:, :1:2].view(xp.int16)`.